src/Illuminate/Routing/Route.php

```diff
@@ -560,6 +560,24 @@ public function domain($domain = null)
         return $this;
     }
 
+    /**
+     * Get the sub domain for the route.
+     *
+     * @param string $subDomain
+     * @param string|null $domain
+     * @return $this
+     */
+    public function subDomain($subDomain, $domain = null)
+    {
+        if (is_null($domain)) {
+            $domain = parse_url(getenv('APP_URL'))['host'];
```

Comment:
```suggestion
            $domain = parse_url(config('app.url'))['host'];
```

Reply:
I have already try that, but an error has returned in my test

```
ReflectionException : Class config does not exist
 D:\lab\project\framework\src\Illuminate\Container\Container.php:779
 D:\lab\project\framework\src\Illuminate\Container\Container.php:658
 D:\lab\project\framework\src\Illuminate\Container\Container.php:609
 D:\lab\project\framework\src\Illuminate\Foundation\helpers.php:121
 D:\lab\project\framework\src\Illuminate\Foundation\helpers.php:283
 D:\lab\project\framework\src\Illuminate\Routing\Route.php:573
 D:\lab\project\framework\tests\Routing\RoutingRouteTest.php:579
```

Reply:
You can't use the `config` Foundation helper in the routing component as the routing component does not depend on Foundation.

Reply:
Yes, but what should I use?

Reply:
You have access to the entire container in the `Router` (`$this->container`).

Reply:
I can not load the data from the environment file, and I do not have access from the container.

Reply:
What do you mean? You can do `$this->container->make('config')->get('app.url')`

Reply:
I have try this, but i have this error in my test

```
1) Illuminate\Tests\Routing\RoutingRouteTest::testRouteSubDomainRegistrationWitheoutDomain
ReflectionException: Class config does not exist

D:\lab\project\framework\src\Illuminate\Container\Container.php:779
D:\lab\project\framework\src\Illuminate\Container\Container.php:658
D:\lab\project\framework\src\Illuminate\Container\Container.php:609
D:\lab\project\framework\src\Illuminate\Routing\Route.php:574
D:\lab\project\framework\tests\Routing\RoutingRouteTest.php:579
```

Reply:
I want to get content of APP_URL variable but I can't retrive this with function env() or getenv()

Reply:
I told you already why that happens in the test. You must bind the `config` service in the test yourself otherwise the container in the test doesn't know how to resolve it and then you get that exception.

Reply:
I am really sorry but I do not know how to define a container. I tried in my test but I do not understand how to do it.

Reply:
Like I said in a previous comment, you can check the `getRouter` method which you are calling in your tests as well to see how it's done.

Basically,

```php
use Illuminate\Config\Repository;

$config = new Repository();
$config->set('app.url', 'http://localhost/');
$container->instance('config', $config);
```

Reply:
Yeah, but where i can add $container->instance ? In my test i not have access to $container

Reply:
```php
$config = new Repository();
$config->set('app.url', 'foo.bar');

$router = $this->getRouter();
....
```

Reply:
Either add the binding on the `getRouter` method or create a similar method which does what the `getRouter` method does but with the addition of binding the config.

Reply:
Yes but how to make this ?

Reply:
At this point I'm not sure if you are trolling or what :)

https://github.com/laravel/framework/blob/39d64c5233b01f736a3eb8ee42dedc2c858e14a0/tests/Routing/RoutingRouteTest.php#L1567

You can just add
```php
$config = new Repository();
$config->set('app.url', 'http://localhost/');
$container->instance('config', $config);
```

inside this method...

Reply:
I'm not trolling really, but not work

```php
protected function getRouter()
{
    $container = new Container;

    $config = new Repository;
    $config->set('app.url', 'http://foo.bar/');

    $container->instance('config', $config);

    $router = new Router(new Dispatcher, $container);

    $container->singleton(Registrar::class, function () use ($router) {
        return $router;
     });

    return $router;
}
```

and my function subDomain:
```php
public function subDomain($subDomain, $domain = null)
{
    dump(config('app.url'));
}
```

error returned by phpunit:
```
1) Illuminate\Tests\Routing\RoutingRouteTest::testRouteSubDomainRegistrationWitheoutDomain
ReflectionException: Class config does not exist

D:\lab\project\framework\src\Illuminate\Container\Container.php:779
D:\lab\project\framework\src\Illuminate\Container\Container.php:658
D:\lab\project\framework\src\Illuminate\Container\Container.php:609
D:\lab\project\framework\src\Illuminate\Foundation\helpers.php:121
D:\lab\project\framework\src\Illuminate\Foundation\helpers.php:283
D:\lab\project\framework\src\Illuminate\Routing\Route.php:575
D:\lab\project\framework\tests\Routing\RoutingRouteTest.php:580
```

Reply:
`dump(config('app.url'));` should be `dump($this->container->make('config')->get('app.url'));`

Reply:
I love you, i'm really stupid today <3

Reply:
Haha, no problem 😄 I'm glad that I was helpful :)